addons/account/models/account_analytic_line.py

```diff
@@ -79,6 +80,17 @@ def on_change_unit_amount(self):
         self.general_account_id = account
         self.product_uom_id = unit
 
+    def write(self, vals):
+        if self.move_line_id:
```

Comment:
Shouldn't we prevent a certain number of fields of being changed (like amount/partner/accounts/...).
I mean if a user decided to change the description of the analytic item, I don't see why we're preventing it. (for example, he created 2 analytic items from an account_move_line and wants to explain what's the difference between them, he can't do so now)

Reply:
I've protected the amount and accounts as they're the important ones, for the other ones like partner/product/date/... I don't know which matter if the user edit them, should we ask tsb?

For the context, I agree it could break some current flows we're not aware of, but I think it is the way to go.
It also how the posted moves are protected: https://github.com/odoo/odoo/blob/e7196ce22446afbb9b735dc44e19e5754c3b7983/addons/account/models/account_move_line.py#L1679-L1680

By grepping the regex `analytic.*unlink`, it looks like the unlink of analytic lines from move lines are covered, unless the analytic lines from stock_move/workorder/project/... can also be related to a move line.

Reply:
I would definitely add the `move_line_id` (to avoid someone adding a new one) and the financial account to the protected fields.
Maybe `date`. But yeah, you should ask tsb to be sure if there are more.

I wasn't aware of the posted moves one, I guess it's fine then

> By grepping the regex analytic.*unlink, it looks like the unlink of analytic lines from amls are covered, unless the analytic lines from stock_move/workorder/project/... can also be related to an aml.

Guess what I did to find the one in the bank_rec_widget :smile: 
Other flows shouldn't have aal with an aml without being in the distribution (or it should be challenged because it would be weird)